### DISCUSSION TITLE
Berry and deepsleep control?

### DISCUSSION THREAD

**Discussion by btsimonh** [arendst/Tasmota#19278] [Q&A]
Has anyone got a Berry sample which puts the ESP32 into deepsleep, and wakes it at a specific time or at specific intervals?  Is this even possible?

I'm thinking it would be nice to deepsleep between taking pictures every hour....

I note the DeepSleep command documentation - does this apply for ESP32 as well?

**Comment by sfromis** [comment]
If the command `DeepSleepTime` (there is no command named just `DeepSleep`) did not apply to ESP32, it would be excluded from the builds. Instead the [driver](https://github.com/arendst/Tasmota/blob/development/tasmota/tasmota_xdrv_driver/xdrv_29_deepsleep.ino) has `#ifdef` for tweaks to ESP32 instead of ESP8266. I'd expect that "someone" has tested and got it to work, but I've not tried it.

And I've not tried the https://tasmota.github.io/docs/ULP/ with Berry either, but the docs page does include coding around wakeup.

**Reply by btsimonh** [reply]
ok, just tried it on ESP32-S3.

Normal TAS command:
`deepsleeptime 120` results in the esp sleeping, waking up every 2 minutes.  
I think all the other items in the docs apply (e.g. the default teleperiod of 300 seconds is a special value which means the ESP will sleep right after connecting to wifi.  If you set `teleperiod 90`, then the ESP will stay awake for ~90s after awakening.

Of course, you could `deepsleeptime 0` from berry at startup, and then set a time again when you want to sleep, however this would cause a flash write.  An alternative would be to join two GPIOs together, and set one as `DeepSleep (182)`, and then control the other from berry.
From the soucre code I can't see another way of preventing sleep without writing flash.

ULP:
The Berry ULP module also calls esp_sleep_enable_timer_wakeup(time).
On my ESP32-S3
```
import ULP
ULP.sleep(30)
```
will sleep the ESP for 30s.
However, it does not go through a TAS shutdown...  so before calling in Berry, you'd need to set any peripherals into low power (e.g. power off cam with new `wcpoweroff` command, if your cam has power control).

**Reply by stefanbode** [reply]
There is a new commit let you use the TIMERS for wakeup process. Documentation here:
https://github.com/tasmota/docs/blob/master/docs/DeepSleep.md

Only in latest DEV branch
